src/AWS.Deploy.Orchestration/SystemCapabilityEvaluator.cs

```diff
@@ -27,105 +28,161 @@ public class SystemCapabilityEvaluator : ISystemCapabilityEvaluator
+        /// <summary>
+        /// If we ran a successful Docker evaluation, this is the timestamp until which that result
+        /// is valid and we will skip subsequent evaluations
+        /// </summary>
+        private DateTime? _dockerDependencyValidUntilUtc = null;
+
+        public SystemCapabilityEvaluator(ICommandLineWrapper commandLineWrapper)
+        {
+            _commandLineWrapper = commandLineWrapper;
         }
 
-        private async Task<DockerInfo> HasDockerInstalledAndRunning()
+        private async Task<DockerInfo> HasDockerInstalledAndRunningAsync()
         {
             var processExitCode = -1;
             var containerType = "";
             var command = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "docker info -f \"{{.OSType}}\"" : "docker info";
 
-            await _commandLineWrapper.Run(
-                command,
-                streamOutputToInteractiveService: false,
-                onComplete: proc =>
-                {
-                    processExitCode = proc.ExitCode;
-                    containerType = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
-                        proc.StandardOut?.TrimEnd('\n') ??
-                            throw new DockerInfoException(DeployToolErrorCode.FailedToCheckDockerInfo, "Failed to check if Docker is running in Windows or Linux container mode.") :
-                        "linux";
-                });
-
-            var dockerInfo = new DockerInfo(processExitCode == 0, containerType);
+            var cancellationTokenSource = new CancellationTokenSource();
+            cancellationTokenSource.CancelAfter(CAPABILITY_EVALUATION_TIMEOUT_MS);
 
-            return dockerInfo;
+            try
+            {
+                await _commandLineWrapper.Run(
+                    command,
+                    streamOutputToInteractiveService: false,
+                    onComplete: proc =>
+                    {
+                        processExitCode = proc.ExitCode;
+                        containerType = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
+                            proc.StandardOut?.TrimEnd('\n') ??
+                                throw new DockerInfoException(DeployToolErrorCode.FailedToCheckDockerInfo, "Failed to check if Docker is running in Windows or Linux container mode.") :
+                            "linux";
+                    },
+                    cancellationToken: cancellationTokenSource.Token);
+
+
+                var dockerInfo = new DockerInfo(processExitCode == 0, containerType);
+
+                return dockerInfo;
+            }
+            catch (TaskCanceledException)
+            {
+                // If the check timed out, treat Docker as not installed
+                return new DockerInfo(false, "");
+            }
         }
 
         /// <summary>
         /// From https://docs.aws.amazon.com/cdk/latest/guide/work-with.html#work-with-prerequisites,
         /// min version is 10.3
```

Comment:
could you please update this comment to either mention the latest version we support (since we've updated it) or remove this version from the comment.

Reply:
Removed it in https://github.com/aws/aws-dotnet-deploy/pull/837/commits/b22ed89708bf1b4e05ec4b89d774f841be431331. This method is just returning the version anyway, the minimum check is done elsewhere.